plantcv/annotate/classes.py

```diff
@@ -158,3 +162,365 @@ class label, by default "default"
         else:
             for (x, y) in self.coords[self.label]:
                 self.ax.plot(x, y, marker='x', c=self.color)
+
+    def _create_pts_mask(self, mask, labelnames):
+        """Fitler a binary mask based on annotations.
+
+        Parameters
+        ----------
+        mask : numpy.ndarray
+            mask for image size
+
+        Returns
+        ----------
+        pts_mask : numpy.ndarray
+            binary mask of annotations
+        """
+        pts_mask = np.zeros(np.shape(mask), np.uint8)
+        # Create points mask from all annotations
+        for names in labelnames:
+            for (x, y) in self.coords[names]:
+                x = int(x)
+                y = int(y)
+                # Draw pt annotations onto a blank mask
+                pts_mask = cv2.circle(pts_mask, (x, y), radius=0, color=(255), thickness=-1)
```

Comment:
Take a look at `itertools`, particularly `itertools.product` I think to reduce nested for loops to one loop. This also applies to dealing with complexity below.

Reply:
Thanks for the feedback! I have simplified this helper function. 

I spent most of this morning testing different ways to loop through all annotation coordinates, but I'm struggling to find a way to avoid the nested `for` loop within the actual `correct_mask` method. The part that I'm not sure how to get around is that the algorithm currently relies on being aware of the corresponding class label called `names`, but I believe it's possible. If you have more suggestions I'm happy to implement them, and I'm also amenable if you want to make changes directly.

Reply:
itertools.product?

Reply:
`itertools.product` is the first way I tried this, and I'm just not sure how to use it in such a way that keeps awareness of the `key` part of the dictionary rather than just gathering all coordinates. Here's the closest I got today: 

```
import itertools

# still a bad product
for instance in itertools.product(counter.coords.keys(), counter.coords.values()):
    print(instance)
```

```
('default', [(398, 416), (1262, 338), (1701, 378), (1806, 1025), (1518, 1505), (556, 1430), (481, 997), (1090, 754)])
('default', [(321, 486), (308, 888), (333, 993), (321, 1193), (338, 1489)])
('default', [(1423, 529), (1721, 399), (1864, 229)])
('germinated', [(398, 416), (1262, 338), (1701, 378), (1806, 1025), (1518, 1505), (556, 1430), (481, 997), (1090, 754)])
('germinated', [(321, 486), (308, 888), (333, 993), (321, 1193), (338, 1489)])
('germinated', [(1423, 529), (1721, 399), (1864, 229)])
('other', [(398, 416), (1262, 338), (1701, 378), (1806, 1025), (1518, 1505), (556, 1430), (481, 997), (1090, 754)])
('other', [(321, 486), (308, 888), (333, 993), (321, 1193), (338, 1489)])
('other', [(1423, 529), (1721, 399), (1864, 229)])
```

I found that this `sum` syntax I've added to `_create_pts_mask` does a nice job of combining all annotations into a single list, but for the algorithm to utilize this format of data, we would also need to create a list of the same length that contains the class label corresponding to each value. This seems possible, I'm just not certain that `itertools.product` is the method that can create this based on the googling I was doing today, but that could be my lack of familiarity with the package.